core/domain/html_validation_service.py

```diff
@@ -880,6 +881,42 @@ def get_filename_with_dimensions(old_filename, exp_id):
     return new_filename
 
 
+def add_math_content_to_math_rte_components(html_string):
+    """Replaces the attribute raw_latex-with-value in all Math RTE tags with
+    a new attribute math_content-with-value. The new attribute has an additional
+    field for storing SVG filenames. The field for SVG filename will be an empty
+    string.
+
+    Args:
+        html_string: str. HTML string to modify.
+
+    Returns:
+        str. Updated HTML string with all Math RTE tags having the new
+        attribute.
+    """
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
+    for math in soup.findAll(name='oppia-noninteractive-math'):
+        if math.has_attr('raw_latex-with-value'):
+            raw_latex = json.loads(unescape_html(math['raw_latex-with-value']))
+            math_content_dict = {
+                'raw_latex': raw_latex,
+                'svg_filename': ''
+            }
+            # Normalize and validate the value before adding to the math tag.
+            normalized_math_content_dict = (
+                objects.MathExpressionContent.normalize(math_content_dict))
+            # Delete the attribute raw_latex-with-value.
+            del math['raw_latex-with-value']
+            # Add the new attribute math_expression_contents-with-value.
+            math['math_content-with-value'] = (
+                escape_html(
+                    json.dumps(normalized_math_content_dict, sort_keys=True)))
+    # We need to replace the <br/> tags (if any) with  <br> because for passing
+    # the textangular migration tests we need to have only <br> tags.
+    return python_utils.UNICODE(soup).replace('<br/>', '<br>')
```

Comment:
Do we need to do the opposite conversion at the top? (see line 151 of this file)

Reply:
I don't think it's needed. that is only needed for conversion to textangular.
Although adding it doesn't affect the result of the function